x-pack/plugins/maps/public/classes/layers/tiled_vector_layer/tiled_vector_layer.tsx

```diff
@@ -89,37 +91,41 @@ export class TiledVectorLayer extends VectorLayer {
   }
 
   _syncSourceBindingWithMb(mbMap: unknown) {
-    // @ts-ignore
-    const mbSource = mbMap.getSource(this.getId());
-    if (!mbSource) {
-      const sourceDataRequest = this.getSourceDataRequest();
-      if (!sourceDataRequest) {
-        // this is possible if the layer was invisible at startup.
-        // the actions will not perform any data=syncing as an optimization when a layer is invisible
-        // when turning the layer back into visible, it's possible the url has not been resovled yet.
-        return;
-      }
+    // @ts-expect-error
+    const mbSource = mbMap.getSource(this._getMbSourceId());
+    if (mbSource) {
+      return;
+    }
```

Comment:
You could execute this condition on line 188. The second two sync calls in that method both return if this condition is met.

Reply:
actually, we can't. those two methods are opposites :) . The `_syncSourceBinding` will check if there's already a source and skip if yes (= no syncing needs to be done). The `_syncStylePropertiesWithMb` one will check if there's a source, and skip if no (= syncing cannot be done).

Reply:
Ha, you're right. Disregard!